packages/fuel-gauge/src/abi/abi-gen.test.ts

```diff
@@ -1,23 +1,60 @@
-import { log } from 'console';
+import { readFileSync } from 'fs';
+import { AbiGen } from 'fuels';
+import { getProgramDetails } from 'fuels/cli-utils';
+import { join } from 'path';
 
 import { AbiProjectsEnum, getAbiForcProject } from './utils';
 
 /**
  * @group node
  */
 describe('AbiGen', () => {
-  test('contract', () => {
-    const { abiContents } = getAbiForcProject(AbiProjectsEnum.ABI_CONTRACT);
-    log(abiContents);
-  });
+  test('AbiGen generates all files correctly', () => {
+    const fixtureResultMap = new Map([
+      ['index', 'index.ts'],
+      ['common', 'common.ts'],
 
-  test('script', () => {
-    const { abiContents } = getAbiForcProject(AbiProjectsEnum.ABI_SCRIPT);
-    log(abiContents);
-  });
+      ['contract-index', 'contracts/index.ts'],
```

Comment:
- Could we seperate out the tests into `contract`, `predicate`, `script` and `common` so that it's easier to deduce whether an error occurs.
- I think having the folder structure match that of the expected output files might be beneficial.
- Would also using the actual file extensions be possible? Rather than `.txt` file extension.
- An auto-update feature would also be handy.

Reply:
1. I intentionally made it like this to verify that only one `common` and `index` file are generated regardless of the amount of programs being typegen'd. If I were to split it up into multiple tests, I'd have to still write a test similar to this one where multiple programs are passed so that I can test this behavior.
2. You're right, I updated it.
3. I used `.txt` to be sure that the fixtures will be ignored by linting, prettifier, etc. The `.hbs` extension was previously used for the fixtures which did the same thing.
4. What do you mean with an auto-update feature? For what?